storage/innobase/srv/srv0srv.cc

```diff
@@ -2967,8 +2967,6 @@ srv_task_execute(void)
 
 		os_atomic_inc_ulint(
 			&purge_sys->pq_mutex, &purge_sys->n_completed, 1);
-
-		srv_inc_activity_count();
```

Comment:
so the old logic bumped the activity twice for the purge.
1) when the workers completed. ie.  number_of_pages from batch/total_worker times.. 
2) in coordinator when the total pages handled by a batch is not zero.

So interestingly, the not so accurate activity count by purge didn't influence the flusher that much..

Reply:
Looks like the main reason for this to become visible is time points when the purge thread becomes active. With upstream patch purge thread will be woken up once a second and at the same time flusher is supposed to clean cache after one second if there is not any activity. As a result we get some kind of a conflict at this point.
Do you think we could get unexpected effects after this fix? Is there anything you could suggest to recheck?

Reply:
AFAIU, the main reason for this counters for purge is not make server think it is idle.
If server thinks, it is idle, it will consider flushing at full io cpacity. and this is bad if there are some other real flushing happening.

So activity counters help server to flush in more adaptive manner (adaptive flushing/ flush_list flushing)

Reply:
@satya-bodapati Both of these calls to `srv_inc_activity_count()` were added by Laurynas as an IMPROVEMENT in this commit
https://github.com/percona/percona-server/commit/05255ba87e77c46cc84b1611af2ab8dc0d32d51f
Worst case scenario, even if we remove this call here, we will be just as good as upstream.
Nevertheless, we can potentially do something similar to what we did in the `srv_purge_coordinator_thread` - call `srv_inc_activity_count()` only if there was actual activity. However, in this case we will have to extend `QUE_NODE_PURGE` queue task and probably its dependent data structure to hold the result of task execution and somehow extract it here from the `thr`

Reply:
@percona-ysorokin it looks to me that the activity counters in purge are totally un-necessary with dedicated LRU threads. At the time of commit 05255ba, there was coupling in page cleaner thread that it has to do LRU flushes from same thread.

Since PS has dedicated LRU flushing, IMHO we can get rid of these.

Also, from the minimal mtr testcase you gave, it does only INSERTs which doesn't add to purge history length. So ideally purge shouldn't have a role here. 

Neverthless, am trying to understand what caused purge to interfere.